manager/src/main/java/org/openremote/manager/rules/JsonRulesBuilder.java

```diff
@@ -658,6 +663,8 @@ protected Condition buildLhsCondition(JsonRule rule, RuleState ruleState) {
 
         return facts -> {
             ruleState.update(timerService::getCurrentTimeMillis);
+            // Execute the actions initially only when the rules engine has already fired and the rule state is matched
+            ruleState.executeActionsWhenMatched = rulesEngine.hasPreviouslyFired() && ruleState.matched;
```

Comment:
Don't think we need this in the ruleState, just use `rulesEngine.hasPreviouslyFired()` directly in the L681

Reply:
Yes that makes sense now that the rules engine keeps track of it. :+1: